graylog2-inputs/src/main/java/org/graylog2/inputs/codecs/GelfChunkAggregator.java

```diff
@@ -221,19 +224,25 @@ public boolean equals(Object o) {
 
             final ChunkEntry that = (ChunkEntry) o;
 
+            if (!id.equals(that.id)) return false;
             if (firstTimestamp != that.firstTimestamp) return false;
-            if (!chunkSlotsWritten.equals(that.chunkSlotsWritten)) return false;
-            //noinspection RedundantIfStatement
-            if (!payloadArray.equals(that.payloadArray)) return false;
+            if (chunkSlotsWritten.get() != that.chunkSlotsWritten.get()) return false;
+            if (!equalsAtomicReferenceArray(this.payloadArray, that.payloadArray)) return false;
```

Comment:
:white_check_mark: Agreed on removing the `Atomic*` instance variables from `equals()` and `hashCode()`.